### PR TITLE
Added .NET framework book

### PR DESCRIPTION
**Pull request by Abinashbunty**
## What does this PR do?
Add resource

### Description
It is book related to .NET

### Why is this valuable (or not)?
.NET is popular for building websites and apps.

### How do we know it's really free?
It's a direct link to the pdf file.

### For book lists, is it a book? For course lists, is it a course? etc.
It is a book.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -307,6 +307,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Diving into ASP.NET WebAPI (2016)](https://github.com/akhilmittal/FreeBooks/) - Akhil Mittal (PDF)
 * [Intro to ASPNET MVC 4 with Visual Studio 2011 Beta (2012)](http://download.microsoft.com/download/0/f/b/0fbfaa46-2bfd-478f-8e56-7bf3c672df9d/intro%20to%20asp.net%20mvc%204%20with%20visual%20studio%20-%20beta.pdf) - Rick Anderson, Scott Hanselman (PDF)
 * [Introducing ASP.NET Web Pages 2 (2012)](https://download.microsoft.com/download/0/F/B/0FBFAA46-2BFD-478F-8E56-7BF3C672DF9D/Introducing%20ASP.NET%20Web%20Pages%202.pdf) - Mike Pope (PDF)
+* [.NET Framework Notes for Professionals](https://books.goalkicker.com/DotNETFrameworkBook/DotNETFrameworkNotesForProfessionals.pdf) (PDF)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:310]: Please add the "Compiled from StackOverflow documentation" attribution we use for goalkicker books

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -1086,7 +1087,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 
 ### JavaScript
 
-* [Airbnb JavaScript Style Guide() {](https://airbnb.io/javascript/) - Airbnb (HTML)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1089]: is this a typo?

**Review by eshellman — CHANGES_REQUESTED** [review]

File: books/free-programming-books-langs.md (hunk 1 above)
Comment (Abinashbunty) [reply books/free-programming-books-langs.md]: Done

File: books/free-programming-books-langs.md (hunk 2 above)
Comment (Abinashbunty) [reply books/free-programming-books-langs.md:1090]: Yeah. The title of the page referred to in the link doesn't have { or (). So I removed them.

**Comment by Abinashbunty** [comment]
@eshellman I updated with changes. 😄

**Comment by eshellman** [comment]
Please excuse the delay. see #9109

**Comment by Abinashbunty** [comment]
@eshellman No worries about the delay. 😄 If needed, I would love to maintain the project and help the team. 👍🏻

**Pull request merged**